docker/docker-compose.yml

```diff
@@ -0,0 +1,32 @@
+version: '3.5'
+
+services:
+  puppet:
+    hostname: puppet.test
+    image: ${PUPPET_TEST_DOCKER_IMAGE:-puppet/puppetserver-standalone}
+    environment:
+      - PUPPETSERVER_HOSTNAME=puppet.test
+      - PUPPERWARE_ANALYTICS_ENABLED=${PUPPERWARE_ANALYTICS_ENABLED:-false}
+      - DNS_ALT_NAMES=puppet,puppet.test,${DNS_ALT_NAMES:-}
+    dns_search: '.test'
+    networks:
+      puppetserver_test:
+        aliases:
+          - puppet.test
+
+  compiler:
+    hostname: compiler.test
+    image: ${PUPPET_TEST_DOCKER_IMAGE:-puppet/puppetserver-standalone}
+    environment:
+      - PUPPERWARE_ANALYTICS_ENABLED=${PUPPERWARE_ANALYTICS_ENABLED:-false}
+      - CA_ENABLED=false
+      - CA_HOSTNAME=puppet.test
+    dns_search: '.test'
+    networks:
+      puppetserver_test:
+        aliases:
+          - compiler.test
+
+networks:
+  puppetserver_test:
```

Comment:
👍 

Even though this adds more to the file, we should probably make this explicit it in other compose files so we have better control of the name.